api/lib/models/project/project-data-mapper-impl.ts

```diff
@@ -0,0 +1,236 @@
+      [AuthLevel.TEAM_MEMBER],
+      undefined,
+      [AuthLevel[AuthLevel.VOLUNTEER]],
+    );
+    super.addRBAC(
+      [this.READ_ALL],
+      [AuthLevel.VOLUNTEER],
+      undefined,
+      [AuthLevel[AuthLevel.PARTICIPANT]],
+    );
+    super.addRBAC(
+      [this.CREATE, this.READ, this.UPDATE],
+      [AuthLevel.PARTICIPANT],
+    );
+  }
+
+  public delete(projectID: UidType): Promise<IDbResult<void>> {
+    const query = squel.delete({ autoQuoteTableNames: true, autoQuoteFieldNames: true })
+      .from(this.tableName)
+      .where(`${this.pkColumnName} = ?`, projectID)
+      .toParam();
+    query.text = query.text.concat(';');
+    return from(
+      this.sql.query(query.text, query.values, { cache: false }),
+    ).pipe(
+      map(() => ({ result: 'Success', data: undefined })),
+    ).toPromise();
+  }
+
+  public get(projectID: UidType, opts?: IUowOpts): Promise<IDbResult<Project>> {
+    let queryBuilder = squel.select({ autoQuoteFieldNames: true, autoQuoteTableNames: true })
+      .from(this.tableName);
+    if (opts && opts.fields) {
+      queryBuilder = queryBuilder.fields(opts.fields);
+    }
+    queryBuilder = queryBuilder
+      .where(`${this.pkColumnName} = ?`, projectID);
+    const query = queryBuilder.toParam();
+    query.text = query.text.concat(';');
+    return from(this.sql.query<Project>(
+      query.text,
+      query.values,
+      { cache: true },
+    ))
+      .pipe(
+        map((event: Project[]) => ({ result: 'Success', data: event[0] })),
+      )
+      .toPromise();
+  }
+
+  public async getAll(opts?: IUowOpts): Promise<IDbResult<Project[]>> {
+    let queryBuilder = squel.select({ autoQuoteFieldNames: true, autoQuoteTableNames: true })
+      .from(this.tableName);
+    if (opts && opts.startAt) {
+      queryBuilder = queryBuilder.offset(opts.startAt);
+    }
+    if (opts && opts.count) {
+      queryBuilder = queryBuilder.limit(opts.count);
+    }
+    if (opts && opts.byHackathon) {
+      queryBuilder = queryBuilder
+        .where(
+          'hackathon = ?',
+          await (opts && opts.hackathon ?
+          Promise.resolve(opts.hackathon) :
+          this.activeHackathonDataMapper.activeHackathon
+            .pipe(map(hackathon => hackathon.uid))
+            .toPromise()),
+        );
+    }
+    const query = queryBuilder.toParam();
+    query.text = query.text.concat(';');
+    return from(this.sql.query<Project>(query.text, query.values, { cache: true },
+      ))
+      .pipe(map((projects: Project[]) => ({ result: 'Success', data: projects })),
+      )
+      .toPromise();
+  }
+
+  public async getByUser(uid: UidType, opts?: IUowOpts): Promise<IDbResult<Project>> {
+    let queryBuilder = squel.select({ autoQuoteTableNames: true, autoQuoteFieldNames: true })
+      .from(this.teamTableName, 'pt')
+      .field('pl.projectName')
+      .field('pt.*')
+      .field('ta.tableNumber')
+      .field('cl.*')
+      .join(this.tableName, 'pl', 'pl.projectID=pl.projectID')
```

Comment:
I know that these abbreviations were originally used, but I think one of the comments back then was to change this to something more readable. Could you change 'pl' to 'project_list', 'pt' to 'project_team' and so on?